### PR TITLE
Fix bin permissions

### PR DESCRIPTION
**Pull request by fisker**
fixes: #6900 , this is a bug introduced in #6200

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: .azure-pipelines/jobs/prod-build.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 steps:
   - template: ../steps/install-nodejs.yml
   - template: ../steps/install-dependencies.yml
-  - script: yarn build
+  - script: yarn build && cd dist && list -l && cd ..
```
Comment (lydell) [inline .azure-pipelines/jobs/prod-build.yml]: Forgot to remove this?

File: scripts/build/build.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ async function run(params) {
     await execa("rm", ["-rf", ".cache"]);
   }
 
-  const bundleCache = new Cache(".cache/", "v18");
+  const bundleCache = new Cache(".cache/", "v19");
```
Comment (lydell) [inline scripts/build/build.js]: Is this related to the issue or something else you found?

File: .azure-pipelines/jobs/prod-build.yml (hunk 1 above)
Comment (fisker) [reply .azure-pipelines/jobs/prod-build.yml]: no, I want show the result on CI, then remove it

File: scripts/build/build.js (hunk 2 above)
Comment (fisker) [reply scripts/build/build.js]: Cache version should always update, when updating build script, I've update this many times.

Comment (fisker) [reply scripts/build/build.js]: I's because build script changes is not calculated in cache, only effect people who run build, like you, haha

**Comment by fisker** [comment]
https://dev.azure.com/prettier/prettier/_build/results?buildId=2884&view=logs&jobId=afd2db75-0a23-56ec-00f6-5aec6de004e8&taskId=bdbf96c4-18ba-5e53-f741-af2442c390ef&lineStart=34&lineEnd=35&colStart=1&colEnd=1

`+x` added

**Pull request merged**